reV/supply_curve/supply_curve.py

```diff
@@ -131,42 +135,253 @@ def _parse_sc_points(sc_points, sc_features=None):
             logger.debug('Adding Supply Curve Features table with columns: {}'
                          .format(sc_features.columns.values.tolist()))
 
+        if 'transmission_multiplier' in sc_points:
+            col = 'transmission_multiplier'
+            sc_points.loc[:, col] = sc_points.loc[:, col].fillna(1)
+
         logger.debug('Final supply curve points table has columns: {}'
                      .format(sc_points.columns.values.tolist()))
 
         return sc_points
 
     @staticmethod
-    def _parse_sc_gids(trans_table, gid_key='sc_gid'):
-        """Filter the trans table, extract unique sc gids, make bool mask.
+    def _get_merge_cols(sc_columns, trans_columns):
+        """
+        Get columns with 'row' or 'col' in them to use for merging
 
         Parameters
         ----------
+        sc_columns : list
+            Columns to search
+        trans_cols
+
+        Returns
+        -------
+        merge_cols : dict
+            Columns to merge on
+        """
+        merge_cols = {}
+        for c_val in ['row', 'col']:
+            trans_col = [c for c in trans_columns if c_val in c]
+            sc_col = [c for c in sc_columns if c_val in c]
+            if trans_col and sc_col:
+                merge_cols[sc_col[0]] = trans_col[0]
+
+        return merge_cols
+
+    @staticmethod
+    def _parse_trans_table(trans_table):
+        """
+        Import transmission features table
+
+        Parameters
+        ----------
+        trans_table : pd.DataFrame | str
+            Table mapping supply curve points to transmission features
+            (either str filepath to table file or pre-loaded dataframe).
+
+        Returns
+        -------
         trans_table : pd.DataFrame
-            reV Supply Curve table joined with transmission features table.
-        gid_key : str
-            Column label in trans_table containing the supply curve points
-            primary key.
+            Loaded transmission feature table.
+        """
+
+        trans_table = parse_table(trans_table)
+
+        drop_cols = ['sc_gid', 'cap_left', 'sc_point_gid']
+        drop_cols = [c for c in drop_cols if c in trans_table]
+        if drop_cols:
+            trans_table = trans_table.drop(columns=drop_cols)
+
+        return trans_table
+
+    @staticmethod
+    def _merge_sc_trans_tables(sc_points, trans_table,
+                               offshore_table=None,
+                               sc_cols=('capacity', 'sc_gid', 'mean_cf',
+                                        'mean_lcoe')):
+        """Merge the supply curve table with the transmission features table.
+
+        Parameters
+        ----------
+        sc_points : pd.DataFrame
+            Table of supply curve point summary
+        trans_table : pd.DataFrame
+            Table mapping supply curve points to transmission features
+        offshore_table : str, optional
```

Comment:
Why is this a str while the other inputs are dataframes?

Reply:
old docstring, fixed